c7n/resources/account.py

```diff
@@ -1279,3 +1279,60 @@ def process(self, resources):
             client.put_public_access_block(
                 AccountId=r['account_id'],
                 PublicAccessBlockConfiguration=config)
+
+
+@filters.register('glue-encryption-enabled')
+class GlueEncryptionEnabled(Filter):
+    """Check if glue SSE-KMS encryption is enabled on the aws account. """
+
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['glue-encryption-enabled']},
+            'CatalogId': {'type': 'string'},
+            'DataCatalogEncryptionSettings': {
+                'type': 'object',
+                'additionalProperties': False,
+                'properties': {
+                    'EncryptionAtRest': {
+                        'type': 'object',
+                        'additionalProperties': False,
+                        'properties': {
+                            'CatalogEncryptionMode': {'type': 'string',
+                            'enum': ['DISABLED', 'SSE-KMS']},
+                            'SseAwsKmsKeyId': {'type': 'string'}
+                        }
+                    },
+                    'ConnectionPasswordEncryption': {
+                        'type': 'object',
+                        'additionalProperties': False,
+                        'properties': {
+                            'ReturnConnectionPasswordEncrypted': {'type': 'boolean'},
+                            'AwsKmsKeyId': {'type': 'string'}
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    permissions = ('glue:GetDataCatalogEncryptionSettings',)
+
+    def process(self, resources, event=None):
+
+        client = local_session(self.manager.session_factory).client('glue')
+
+        try:
+            encryption_setting = client.get_data_catalog_encryption_settings().get(
+                'DataCatalogEncryptionSettings')
+
+        except ClientError as e:
+            if e.response['Error']['Code'] != 'EntityNotFoundException':
+                raise
+
+        if encryption_setting['EncryptionAtRest'].get('CatalogEncryptionMode') != 'DISABLED':
```

Comment:
there's a lot of values in the schema that are not being checked, rather than over specifying here it would be better to allow either a multi-attribute value filter, or just allow it as a single attribute value filter with cached annotation on the encryption setting.